packages/compass-e2e-tests/helpers/compass.ts

```diff
@@ -75,70 +79,27 @@ export const MONGODB_TEST_SERVER_PORT = Number(
   process.env.MONGODB_TEST_SERVER_PORT ?? 27091
 );
 
-export const DEFAULT_CONNECTION_STRING_1 = `mongodb://127.0.0.1:${MONGODB_TEST_SERVER_PORT}/test`;
+export const DEFAULT_CONNECTION_STRING_1 = DEFAULT_CONNECTION_STRINGS[0];
 // NOTE: in browser.setupDefaultConnections() we don't give the first connection an
 // explicit name, so it gets a calculated one based off the connection string
-export const DEFAULT_CONNECTION_NAME_1 = connectionNameFromString(
-  DEFAULT_CONNECTION_STRING_1
-);
+export const DEFAULT_CONNECTION_NAME_1 = DEFAULT_CONNECTION_NAMES[0];
 
 // for testing multiple connections
-export const DEFAULT_CONNECTION_STRING_2 = `mongodb://127.0.0.1:${
-  MONGODB_TEST_SERVER_PORT + 1
-}/test`;
+export const DEFAULT_CONNECTION_STRING_2 = DEFAULT_CONNECTION_STRINGS[1];
 // NOTE: in browser.setupDefaultConnections() the second connection gets given an explicit name
-export const DEFAULT_CONNECTION_NAME_2 = 'connection-2';
-
-export function updateMongoDBServerInfo() {
-  try {
-    const { stdout, stderr } = crossSpawn.sync(
-      'npm',
-      [
-        'run',
-        '--silent',
-        /**
-         * The server info update is done through a separate script and not by
-         * using a MongoClient directly because doing so causes an unexplainable
-         * segfault crash in e2e-coverage task in evergreen CI. Moving this
-         * logic to a separate script seems to solve this problem, but if at any
-         * point the issue returns, feel free to revert this whole change
-         **/
-        'server-info',
-        '--',
-        '--connectionString',
-        `mongodb://127.0.0.1:${String(MONGODB_TEST_SERVER_PORT)}`,
-      ],
-      { encoding: 'utf-8' }
-    );
-    if (stderr?.length) {
-      throw new Error(stderr);
-    }
-    const { version, enterprise } = JSON.parse(stdout);
-    MONGODB_VERSION = version;
-    MONGODB_USE_ENTERPRISE = enterprise ? 'yes' : 'no';
-    debug(
-      `Got server info: v${String(version)} (${
-        enterprise ? 'enterprise' : 'community'
-      })`
-    );
-  } catch (err) {
-    (err as Error).message =
-      'Failed trying to get MongoDB server info:\n\n' + (err as Error).message;
-    throw err;
-  }
-}
+export const DEFAULT_CONNECTION_NAME_2 = DEFAULT_CONNECTION_NAMES[1];
 
 export const serverSatisfies = (
   semverCondition: string,
   enterpriseExact?: boolean
 ) => {
+  const { version, enterprise } = DEFAULT_CONNECTIONS_SERVER_INFO[0];
```

Comment:
Yes it feels safe and indeed only became obvious because of the restructure. We indeed only have multiple connections since recently just for the sake of two. Only thing I can think of is that someone one day might be running servers of multiple different versions at once, but then this will quickly become an obvious bug they will find.

Just thinking that we could start multiple servers of different versions then for the handful of cases where we do care about different results for different versions we could wrap the test in a for loop. Then we probably don't ever have to run the entire e2e test suite multiple times in CI which could alleviate some of the limited mac resource problems.. 

Anyway. That's just me thinking out loud. I don't think we need to do anything right now. And there might be an opportunity here in future :)